src/cpp/src/sampler.hpp

```diff
@@ -22,7 +22,7 @@
 
 namespace ov::genai {
 // Modifyed Knuth–Morris–Pratt algorithm which returns tokens following after every needle occurance in haystack
-std::vector<int64_t> kmp_search(const std::vector<int64_t>& haystack, const std::vector<int64_t>& needle) {
+inline std::vector<int64_t> kmp_search(const std::vector<int64_t>& haystack, const std::vector<int64_t>& needle) {
```

Comment:
I think we just need to move all these methods to .cpp and keep only Sampler class declaration in .hpp file